process/executionTrack/executionResultsTracker_test.go

```diff
@@ -0,0 +1,207 @@
+package executionTrack
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/multiversx/mx-chain-core-go/data/block"
+	"github.com/multiversx/mx-chain-go/testscommon"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewExecutionResultsTracker(t *testing.T) {
+	t.Parallel()
+
+	tracker, err := NewExecutionResultsTracker()
+	require.NoError(t, err)
+	require.False(t, tracker.IsInterfaceNil())
+}
+
+func TestSetAndGetLastNotarizedResult(t *testing.T) {
+	t.Parallel()
+
+	tracker, _ := NewExecutionResultsTracker()
+
+	_, err := tracker.GetLastNotarizedExecutionResult()
+	require.Equal(t, ErrNilLastNotarizedExecutionResult, err)
+
+	err = tracker.SetLastNotarizedResult(nil)
+	require.Equal(t, ErrNilExecutionResult, err)
+
+	execResult := &block.ExecutionResult{
+		Nonce: 10,
+	}
+	err = tracker.SetLastNotarizedResult(execResult)
+	require.NoError(t, err)
+
+	lastNotarizedResult, err := tracker.GetLastNotarizedExecutionResult()
+	require.NoError(t, err)
+	require.Equal(t, execResult, lastNotarizedResult)
+}
+
+func TestAddExecutionResult_AllBranches(t *testing.T) {
+	t.Parallel()
+
+	tracker, _ := NewExecutionResultsTracker()
```

Comment:
can you add a t.run for each tested case, instead of comment in the code?

Reply:
done